python/aibrix_kvcache/aibrix_kvcache/l2/connectors/infinistore.py

```diff
@@ -72,52 +72,67 @@ def from_envs(
             )
             return cls(config, conn_id, executor)
 
-        # RDMA
-        addr_family = (
-            AddrFamily.AF_INET if ipv4(host_addr) else AddrFamily.AF_INET6
-        )
-        gid_type = GIDType.ROCE_V2 if link_type != "IB" else GIDType.IB_ROCE_V1
-        if len(dev_list) == 0:
-            logger.info(
-                "AIBRIX_KV_CACHE_OL_INFINISTORE_VISIBLE_DEV_LIST is not set, "
-                "trying to auto-detect visible devices"
-            )
-            addr_range = envs.AIBRIX_KV_CACHE_OL_TRANSPORT_RDMA_ADDR_RANGE
-            rdma = RDMATransport(
-                addr_range=addr_range,
-                addr_family=addr_family,
-                gid_type=gid_type,
+        if HAS_RDMA_TRANSPORT_SUPPORT:
```

Comment:
seems the primary change is 
1. import package on demand
2. add feature flag `HAS_RDMA_TRANSPORT_SUPPORT`. 

the core discovery logic has not changed a lot. 

overall looks good to me